dispatches/case_studies/renewables_case/utils.py

```diff
@@ -0,0 +1,341 @@
+#################################################################################
+# DISPATCHES was produced under the DOE Design Integration and Synthesis Platform
+# to Advance Tightly Coupled Hybrid Energy Systems program (DISPATCHES), and is
+# copyright (c) 2020-2023 by the software owners: The Regents of the University
+# of California, through Lawrence Berkeley National Laboratory, National
+# Technology & Engineering Solutions of Sandia, LLC, Alliance for Sustainable
+# Energy, LLC, Battelle Energy Alliance, LLC, University of Notre Dame du Lac, et
+# al. All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. Both files are also available online at the URL:
+# "https://github.com/gmlc-dispatches/dispatches".
+#################################################################################
+import copy
+import os
+import pathlib
+import pandas as pd
+import numpy as np
+import re
+import json
+
+'''
+prescient_output_to_df, get_gdf and summarize_results are modified from Ben's fork. 
+Run summarize_results will give us the csv file of the simulation including DA dispatch, DA LMP, RT dispatch and RT LMP (like the sweep results)
+
+summarize_revenue is modified from Xian's fork. This function return a dictionary of DA dispatch, DA revenue, RT dispatch and RT revenue and total revenue.
+
+summarize_H2_revenue is to calculate the total hydrogen revenue.
+'''
+
+
+def prescient_output_to_df(file_name):
+    '''Helper for loading data from Prescient output csv.
+        Combines Datetimes into single column.
+    '''
+    df = pd.read_csv(file_name)
+    df['Datetime'] = \
+        pd.to_datetime(df['Date']) + \
+        pd.to_timedelta(df['Hour'], 'hour') + \
+        pd.to_timedelta(df['Minute'], 'minute')
+    df.drop(columns=['Date','Hour','Minute'], inplace=True)
+    # put 'Datetime' in front
+    cols = df.columns.tolist()
+    cols = cols[-1:]+cols[:-1]
+    
+    return df[cols]
+
+
+def get_gdf(directory, generator_file_name, generator_name, dispatch_name):
+    gdf = prescient_output_to_df(os.path.join(directory, generator_file_name))
+    gdf = gdf[gdf["Generator"] == generator_name][["Datetime", dispatch_name, dispatch_name + " DA"]]
+    gdf.set_index("Datetime", inplace=True)
+    gdf.rename(columns={ dispatch_name : generator_name + " Dispatch", dispatch_name + " DA" : generator_name + " Dispatch DA"}, inplace=True)
+
+    return gdf
+
+
+def summarize_results(base_directory, generator_name, bus_name, output_directory):
+    """

```

Comment:
This is a good explain of a complete doc string. Please do this for ALL functions in this PR.